include/envoy/network/connection.h

```diff
@@ -261,6 +261,18 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * @return std::chrono::milliseconds The delayed close timeout value.
    */
   virtual std::chrono::milliseconds delayedCloseTimeout() const PURE;
+
+  /**
+   * Set the order of the write filters, indicating whether it is reversed to the filter chain
+   * config.
+   */
+  virtual void setWriteFilterOrder(bool reversed) PURE;
+
+  /**
+   * @return bool indicates whether write filters should be in the reversed order of the filter
+   *         chain config.
+   */
+  virtual bool reverseWriteFilterOrder() const PURE;
```

Comment:
A few things:

1) As I small optimization, I think you can send reversed directly into `FilterManagerImpl` and then you won't need a public interface getter? 
2) On all of the interface changes, can you add a TODO to remove with a link to this PR and an explanation that the function is deprecated?
3) Another way of doing this that is slightly cleaner would be to actually take the bool as part of the constructor constructor, but that would then involve dispatcher interface changes, so I'm not sure it's that much better. This seems fine for something we are going to turn around and revert anway.

Reply:
1. IMO, sending reversed info directly to FilterManagerImpl will make its APIs less consistent. for instance, we will need to pass bool reversed_write_filter_order to addFilter and addWriteFilter, but not addReadFilter. The Connection interface has only 2 subclasses, I think it is fine to provide this public API reverseWriteFilterOrder().

2. I will add TODOs once we finalize the optimization, in particular, #1. 

3. Yeah, I would prefer the current way. We are implementing something to be deprecated soon and either way requires similar amount of interface changes.

Reply:
For (1) I was suggesting just adding a setFilterWriteOrder() on FilterManagerImpl concrete class (not interface), but honestly it doesn't matter when we are going to turn around and delete all of this soon. I would just add TODO/deprecation text so that we can clean this up (also please add a notice to deprecated.md about this change also).

Reply:
Hmm, sorry, I thought we were going to default both to true (reversed) and then just remove the option? IMO this is what we should do since I strongly suspect if no one complained about this yet no one will care, since the current ordering makes no sense. Thoughts on making the defaults for network and HTTP true?

Reply:
Setting to default true makes sense to me, also for faster deprecation.

@htuch suggested to false as the default value to give longer grace period to OSS Envoy users. 

I am going to set the default true for both encode and write filters. @htuch, I can revert my change if there is any concerns.

Reply:
Sure, SGTM. +1 on skipping the virtual interface change if you can just do this on the concrete class as well.